### PR TITLE
[fuzz] fix timeouts in xDS fuzzer

### PR DESCRIPTION
**Pull request by samflattery**
Signed-off-by: Sam Flattery <samflattery@google.com>

Commit Message: Fix timeouts in xDS fuzzer
Additional Description:

- xDS fuzzer was timing out frequently on OSS fuzz while waiting for ACKs, so I added more wait calls which fixed the issues
- added corpus entries that crashed on OSS fuzz
- I'm still not sure whether routes can be removed or not, according to Harvey here (https://github.com/envoyproxy/envoy/pull/7543#discussion_r302973242), routes cannot be removed in SOTW updates, and after looking through the traces for delta updates, I see things similar to `ignoring delta request to remove resource route_0`, so it doesn't seem to be possible?  

Risk Level: Low
Testing: Fixes crashes on OSS fuzz
Docs Changes: N/A
Release Notes: N/A
Fixes: 
- https://oss-fuzz.com/testcase-detail/5121759976882176
- https://oss-fuzz.com/testcase-detail/5181759124930560

### CODE REVIEW

**Comment by samflattery** [comment]
@asra is the tcp_proxy_integration_test flaky? It seems to pass locally for me but fails here on Windows

File: test/server/config_validation/xds_fuzz.cc [hunk 1]
```diff
@@ -189,12 +189,16 @@ void XdsFuzzTest::replay() {
   // URL so just don't check them until a listener is added
   bool sent_listener = false;
 
+  uint32_t added = 0;
```
Comment (asraa) [inline test/server/config_validation/xds_fuzz.cc:192]: Are these just for future accounting?

File: test/server/config_validation/xds_fuzz.cc [hunk 2]
```diff
@@ -203,47 +207,59 @@ void XdsFuzzTest::replay() {
...
     case test::server::config_validation::Action::kRemoveRoute: {
-      if (sotw_or_delta_ == Grpc::SotwOrDelta::Sotw) {
-        // routes cannot be removed in SOTW updates
-        break;
-      }
+      // it seems like routes cannot be removed
+      // leaving a route out of an SOTW request does not remove it and sending a
+      // remove message in a delta request is ignored
+      ENVOY_LOG_MISC(info, "Ignoring request to remove route_{}",
+                     action.remove_route().route_num());
```
Comment (asraa) [inline test/server/config_validation/xds_fuzz.cc:256]: extremely minor nitty question for readability: were the line wraps in this block necessary?

File: test/server/config_validation/xds_fuzz.cc [hunk 3]
```diff
@@ -203,47 +207,59 @@ void XdsFuzzTest::replay() {
...
+      break;
 
-      auto removed = removeRoute(action.remove_route().route_num());
+      // TODO(samflattery): remove if it's true that routes cannot be removed
```
Comment (asraa) [inline test/server/config_validation/xds_fuzz.cc:259]: might be worth asking on #xds

Comment (asraa) [reply test/server/config_validation/xds_fuzz.cc:259]: (non-blocking, just suggestion)

File: test/server/config_validation/xds_fuzz.cc (hunk 1 above)
Comment (samflattery) [reply test/server/config_validation/xds_fuzz.cc:192]: I used them below on lines 212 and 215 to wait for the updates to come into effect

File: test/server/config_validation/xds_fuzz.cc (hunk 3 above)
Comment (samflattery) [reply test/server/config_validation/xds_fuzz.cc:259]: will do!

File: test/server/config_validation/xds_fuzz.cc (hunk 2 above)
Comment (samflattery) [reply test/server/config_validation/xds_fuzz.cc:256]: the log one goes over the 100 line limit, cleaned up the comment, thanks for catching!

File: test/server/config_validation/xds_fuzz.cc (hunk 1 above)
Comment (asraa) [reply test/server/config_validation/xds_fuzz.cc:192]: Ahhh sorry I missed the expectation compare. thanks!

**Pull request merged**